src/io/flutter/FlutterInitializer.java

```diff
@@ -5,15 +5,122 @@
 import org.jetbrains.annotations.NotNull;
 
+import javax.swing.event.HyperlinkEvent;
+import java.util.UUID;
+
 public class FlutterInitializer implements StartupActivity {
+  private static final String analyticsClientIdKey = "io.flutter.analytics.clientId";
+  private static final String analyticsOptOutKey = "io.flutter.analytics.optOut";
+  private static final String analyticsToastShown = "io.flutter.analytics.toastShown";
+
+  private static Analytics analytics;
+
+  public static Analytics getAnalytics() {
+    if (analytics == null) {
+      final PropertiesComponent properties = PropertiesComponent.getInstance();
+
+      String clientId = properties.getValue(analyticsClientIdKey);
+      if (clientId == null) {
+        clientId = UUID.randomUUID().toString();
+        properties.setValue(analyticsClientIdKey, clientId);
+      }
+
+      final IdeaPluginDescriptor descriptor = PluginManager.getPlugin(PluginId.getId("io.flutter"));
+      assert descriptor != null;
+      analytics = new Analytics(clientId, descriptor.getVersion());
+
+      // Set up reporting prefs.
+      analytics.setCanSend(getCanReportAnalytics());
+
+      // Send initial loading hit.
+      analytics.sendScreenView("main");
+    }
+
+    return analytics;
+  }
+
+  public static void setCanReportAnalaytics(boolean canReportAnalaytics) {
+    if (getCanReportAnalytics() != canReportAnalaytics) {
+      final boolean wasReporting = getCanReportAnalytics();
+
+      final PropertiesComponent properties = PropertiesComponent.getInstance();
+      properties.setValue(analyticsOptOutKey, !canReportAnalaytics);
+      if (analytics != null) {
+        analytics.setCanSend(getCanReportAnalytics());
+      }
+
+      if (!wasReporting && canReportAnalaytics) {
+        getAnalytics().sendScreenView("main");
+      }
+    }
+  }
+
+  public static boolean getCanReportAnalytics() {
+    final PropertiesComponent properties = PropertiesComponent.getInstance();
+    return !properties.getBoolean(analyticsOptOutKey, false);
+  }
+
+  public static void sendActionEvent(AnAction action) {
+    getAnalytics().sendEvent("intellij", action.getClass().getSimpleName());
+  }
 
   @Override
   public void runActivity(@NotNull Project project) {
     FlutterDaemonService.getInstance(project);
+
+    final PropertiesComponent properties = PropertiesComponent.getInstance();
+    if (!properties.getBoolean(analyticsToastShown)) {
+      properties.setValue(analyticsToastShown, true);
+
+      final Notification notification = new Notification(
+        FlutterErrors.FLUTTER_NOTIFICATION_GOUP_ID,
+        "Welcome to Flutter!",
+        "The Flutter plugin anonymously reports feature usage statistics and basic crash reports to " +
+        "Google in order to help Google contribute improvements to Flutter over time. See Google's privacy " +
+        "policy: <a href=\"url\">www.google.com/intl/en/policies/privacy</a>.<br><br>" +
+        "Send anonymous usage statistics?",
```

Comment:
I don't always i18n strings here; I suppose we should? I'll externalize this (and for following ones).